cogboard-webapp/src/components/AddCredential.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useDispatch } from 'react-redux';
+
+import { useToggle } from '../hooks';
+
+import { IconButton, Button } from '@material-ui/core';
+import { Add } from '@material-ui/icons';
+import AppDialog from './AppDialog';
+import CredentialForm from './CredentialForm';
+import { saveCredential } from '../actions/thunks';
+
+const AddCredential = ({ largeButton }) => {
+  const [dialogOpened, openDialog, handleDialogClose] = useToggle();
+  const dispatch = useDispatch();
+
+  const handleAddEndpointClick = () => {
+    openDialog();
+  };
+
+  const handleSubmit = values => {
+    delete values.passwordConfirmation;
+
+    dispatch(saveCredential(values));
+    handleDialogClose();
+  };
+
+  return (
+    <>
+      {largeButton ? (
```

Comment:
Could you extract this ternary statement to a separate 'something'? Call me old fashioned, but I feel this is a case for a normal `if`.